# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/BitWriter.h
# Commit: 412f97da1656
# Full Hash: 412f97da16567da3a0c2a6adc6dbde27a28dea22
# Author: alwu <alwu@mozilla.com>
# Date: 2023-10-11 09:22:03
# Regressor Bug: 1852333
# File Overlap Count: 4
# Description:
#   Bug 1852333 - part2 : detect config change for HEVC stream. r=media-playback-reviewers,padenot
#   
#   This patch detects inband change for the HEVC playback so that we can
#   recreate decoder if the new config is different from the previous one.
#   
# ==============================================================================

diff -r b15b2e7c785a -r 412f97da1656 dom/media/BitWriter.h
--- a/dom/media/BitWriter.h	Wed Oct 11 00:57:39 2023 +0000
+++ b/dom/media/BitWriter.h	Wed Oct 11 00:57:40 2023 +0000
@@ -29,6 +29,12 @@
   // Write RBSP trailing bits.
   void CloseWithRbspTrailing();
 
+  // Advance position forward without modifying buffer, which is usually used
+  // along with the case when directly appending a byte array to the
+  // MediaByteBuffer for the efficiency, instead of writing bits one by one.
+  // So this can only be called when the bit index is zero.
+  void AdvanceBytes(uint32_t aByteOffset);
+
   // Return the number of bits written so far;
   size_t BitCount() const { return mPosition * 8 + mBitIndex; }
 